pkg/kubectl/cmd/drain.go

```diff
@@ -282,16 +283,28 @@ func (o *DrainOptions) getController(sr *api.SerializedReference) (interface{},
 	return nil, fmt.Errorf("Unknown controller kind %q", sr.Reference.Kind)
 }
 
+func newSerializedControllerRef(pod api.Pod) *api.SerializedReference {
+	return &api.SerializedReference{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "SerializedReference",
+		},
+		Reference: api.ObjectReference{ //currently only set based on first controller
+			APIVersion: pod.GetOwnerReferences()[0].APIVersion,
+			Kind:       pod.GetOwnerReferences()[0].Kind,
+			Name:       pod.GetOwnerReferences()[0].Name,
+			Namespace:  pod.GetObjectMeta().GetNamespace(),
+		},
+	}
+}
+
 func (o *DrainOptions) getPodCreator(pod api.Pod) (*api.SerializedReference, error) {
-	creatorRef, found := pod.ObjectMeta.Annotations[api.CreatedByAnnotation]
-	if !found {
+	if controller.GetControllerOf(pod.GetObjectMeta()) == nil {
 		return nil, nil
 	}
 	// Now verify that the specified creator actually exists.
-	sr := &api.SerializedReference{}
-	if err := runtime.DecodeInto(o.Factory.Decoder(true), []byte(creatorRef), sr); err != nil {
-		return nil, err
-	}
+	sr := newSerializedControllerRef(pod)
```

Comment:
Yes, i think it is a better way than translating `ControllerRef` back to `SerializedReference`. I will work on modifying `getController()` and `getPodCreator()` callers.